docs/dev/design/redirects.rst

```diff
@@ -0,0 +1,318 @@
+  This implies that the URL to be matched must have the exact same query arguments,
+  it can't have more or less.
+
+  I believe the implementation described here is the same being used by Netlify,
+  since they have that same restriction.
+
+      If the URL contains other parameters in addition to or instead of id, the request doesn't match that rule.
+
+      https://docs.netlify.com/routing/redirects/redirect-options/#query-parameters
+
+- At the Python level.
+  If done at the DB level,
+  we would need to have a different field
+  with just the path, and other with query arguments.
+
+  The matching of the path would be done at the DB level,
+  and the matching of the query arguments would be done at the Python level.
+  Here we can be more flexible, allowing any query arguments in the matched URL.
+
+  We had some performance problems in the past,
+  but I believe it was mainly due to the use of regex instead of using string operations.
+  And matching the path is still done at the DB level.
+  We could limit the number of redirects that can be created with query arguments,
+  or the number of redirects in general.
+
+Don't run redirects on domains from pull request previews
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We currently run redirects on domains from pull request previews,
+this is a problem when moving a whole project to a new domain.
+
+Do we have the need to run redirects on external domains?
+They are suppose to be temporary domains.
+
+Normalize paths with trailing slashes
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Currently, if users want to redirect a path with a trailing slash and without it,
+they need to create two separate redirects (``/page/`` and ``/page``).
+
+We can simplify this by normalizing the path before matching it.
+
+For example:
+
+- From: ``/page/``
+  To: ``/new/page``
+
+The from path will be normalized to ``/page``,
+and the filename to match will also be normalized before matching it.
+This is similar to what Netlify does:
+https://docs.netlify.com/routing/redirects/redirect-options/#trailing-slash.
+
+Page and exact redirects without a wildcard at the end will be normalized,
+all other redirects need to be matched as is.
+
+Improving exact redirects
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+- Explicitly place the ``$rest`` placeholder in the target URL,
+  instead of adding it automatically.
+
+  Some times users want to redirect to a different path,
+  we have been adding a query parameter in the target URL to
+  prevent the old path from being added in the final path.
+  For example ``/new/path/?_=``.
+
+  Instead of adding the path automatically,
+  users have to add the ``$rest`` placeholder in the target URL.
+  For example:
+
+  - From: ``/old/path/$rest``
+    To: ``/new/path/$rest``
+
+  - From: ``/old/path/$rest``
+    To: ``/new/path/?page=$rest&foo=bar``
+
+- Per-domain redirects.
+  Do users have the need for this?
```

Comment:
We wanted this on EA to be able to redirect `ethicalads.io` to `www.ethicalads.io`. I think it is useful to be able to specify the domain in the redirect itself?

Reply:
But wasn't the main problem that it was redirecting from PR builds? If we don't apply redirects from PR previews, would be needed to have per-domain redirects?

Reply:
I think we may want to look into fixing/improving how canonical domains work before introducing this feature in redirects. If we always redirect to the canonical domain, we won't be needing this "feature"/hack.

Reply:
Agreed. 👍